docs/guides/Sign_Git_commits.md

```diff
@@ -0,0 +1,217 @@
+
+```shell
+gpg --delete-secret-keys $ID
+gpg --delete-keys $ID
+```
+
+### Configure Git
+
+Use the [following commands](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-gpg-key) to set your default signing key in Git to the ID of the GPG key you generated. Replace `$ID` with your actual GPG key ID from the script above.
+
+  ```shell
+  git config --global user.signingkey $ID
+  ```
+
+Then enable automatic signing of Git commits by running:
+
+```shell
+git config --global commit.gpgsign true
+```
+
+### Configure GitHub
+
+To [add your GPG public key to your GitHub account](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account) follow these steps:
+
+1. Navigate to your GitHub account settings.
+2. From the sidebar, click on "**SSH and GPG keys**".
+3. Click on the "**New GPG key**" button.
+4. In the "**Title**" field, enter a descriptive name for the key, like "My GitHub signing key".
+5. Copy the contents of your public key file and paste it into the "**Key**" field.
+
+   ```shell
+   cat $file.gpg-key.pub
+   ```
+
+6. Click "**Add GPG key**" to save.
+
+After completing these steps, your new signing key will be listed in the "**SSH and GPG keys**" section of your GitHub profile.
+
+### Troubleshooting
+
+If you receive the error message "error: gpg failed to sign the data", make sure you added `export GPG_TTY=$(tty)` to your `~/.zshrc` or `~/.bashrc`, and restarted your terminal.
+
+```shell
+sed -i '/^export GPG_TTY/d' ~/.exports
+echo "export GPG_TTY=\$TTY" >> ~/.exports
+```
+
+### Additional settings
+
+Configure caching git commit signature passphrase for 3 hours
+
+```shell
+source ~/.zshrc # or ~/.bashrc
+mkdir -p ~/.gnupg
+sed -i '/^pinentry-program/d' ~/.gnupg/gpg-agent.conf 2>/dev/null ||:
+echo "pinentry-program $(whereis -q pinentry)" >> ~/.gnupg/gpg-agent.conf
+sed -i '/^default-cache-ttl/d' ~/.gnupg/gpg-agent.conf
+echo "default-cache-ttl 10800" >> ~/.gnupg/gpg-agent.conf
+sed -i '/^max-cache-ttl/d' ~/.gnupg/gpg-agent.conf
+echo "max-cache-ttl 10800" >> ~/.gnupg/gpg-agent.conf
+gpgconf --kill gpg-agent
+git config --global credential.helper cache
+#git config --global --unset credential.helper
+```
+
+## Signing commits using SSH
+
+### Generate SSH key
+
+You should not do this if you already have GPG signing set up. One or the other is fine, but not both.
+
+If you do not already have SSH key access set up on your GitHub account, first [generate a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent). To create a new SSH key, you need to run the following command. This will generate a new SSH key of the type `ed25519` and associate it with your email address. Please replace your.name@email with your actual email address.
+
+```shell
+ssh-keygen -t ed25519 -C "your.name@email"
```

Comment:
If we're going to prompt people towards separate signing and access keys, we might as well skip the rename below and just use the `github_signing_key` name here.

```suggestion
ssh-keygen -t ed25519 -C "your.name@email" -f "~/.ssh/github_signing_key"
```

Reply:
done! thank you